flexget/plugins/sites/newznab.py

```diff
@@ -1,125 +1,655 @@
+        return url_param
+
+    def _build_series_ep_season_url_fragment(self, entry):
+        url_param = None
+        if 'series_episode' in entry:
+            url_param = '&ep=%s' % entry['series_episode']
+            # TODO @Andy: test if indexer support episode only search!
+            if 'series_season' in entry:
+                url_param += '&season=%s' % entry['series_season']
+        return url_param
+
+    def _get_custom_query_string(self, entry, task, config):
+        if config.get('custom_query'):
+            try:
+                render_string = entry.render(config['custom_query'])
+                render_string = render_string.strip()
+                if render_string:
+                    return render_string
+            except RenderError as ex:
+                log.error('Could not render custom_query string for: %s error: %s', entry['title'], ex)
+        return None
+
+    # API search entry
+    def search(self, task, entry, config):
+        config = self.prepare_config(config)
+
+        # first check for custom_query
+        custom_query_string = None
+        if config.get('custom_query'):
+            custom_query_string = self._get_custom_query_string(entry, task, config)
+            if not custom_query_string:
+                entry = self.update_metadata(entry, config)  # update metadata and try again
+                custom_query_string = self._get_custom_query_string(entry, task, config)
+                if not custom_query_string:
+                    log.error('Could not build custom_query string for: %s ', entry['title'])
+                    return []  # no fallback logic!
+
+        meta_url = None
+        ep_url = None
+        # build metadata url fragments
+        if 'use_metadata' in config:
+            meta_url = self._build_metaid_url_fragment(entry)
+            if meta_url is None:
+                entry = self.update_metadata(entry, config)
+                meta_url = self._build_metaid_url_fragment(entry)  # try again
+            if self._is_tv_entry(entry) is True:
+                ep_url = self._build_series_ep_season_url_fragment(entry)
+
+        query_list = []
+        # build meta and ep fragment
+        url = self._build_base_url(entry, config)
+        if meta_url and self._is_tv_entry(entry) is True and ep_url:
+            url += '&t=tvsearch' + meta_url + ep_url
+        elif meta_url and self._is_tv_entry(entry) is False:
+            url += '&t=movie' + meta_url
+        else:
+            url += '&t=search'
+            # use existing search_strings
+            query_list = entry.get('search_strings', default=[entry['title']])
+            list_append_unique(query_list, entry['title'], caseinsensitive=True)  # we trust external list?
+
+        # use custom query in all cases (not 100% well defined behaviour in metaid cases)
+        list_insert_unique(query_list, 0, custom_query_string, caseinsensitive=True)
```

Comment:
Its no so much that i want to encode the type, but its easier to use the typename instead of coming up with a alternative name. So in this case i just wanted to distinguish between the query text input, the parsed version, the url_fragment and the "collection" object. So sure i could have named query_list/string also query_text/collection. If i could load python syntax into my brain, as easy as i can switch in the IDE than this would be no problem :)